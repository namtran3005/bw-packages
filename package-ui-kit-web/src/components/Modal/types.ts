import { BackdropProps as MuiBackdropProps } from '@material-ui/core/Backdrop';
import { DialogProps as MuiDialogProps } from '@material-ui/core/Dialog';
import { TransitionProps as MuiTransitionProps } from '@material-ui/core/transitions/transition';

import { ColorsType } from '@bitwala-cryptobank-squad/package-theme';

import { CSSProp } from 'styled-components';
import { ReactNode } from 'react';
import { IconButtonProps } from '../IconButton';

type Docked = 'left' | 'right' | 'middle';

export interface TransitionProps extends MuiTransitionProps {
  docked: Docked;
}

export interface ModalProps
  extends Omit<MuiDialogProps, 'propTypes' | 'TransitionProps' | 'title'> {
  open: boolean;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children?: any;
  paperForeground?: ColorsType;
  docked?: Docked;
  onClose?: IconButtonProps['onClick'];
  position?: 'left' | 'right' | 'middle';
  ariaLabel?: string;
  rightHeaderElement?: React.ReactElement | React.ReactElement[];
  showDismissButton?: boolean;
  maxPaperWidth?: CSSProp;
  title?: ReactNode;
  doNotRestrictTitleWidth?: boolean;
  doNotStretchContentHeight?: boolean;
}

export interface BackdropProps extends MuiBackdropProps {
  open: boolean;
  background?: ColorsType;
}

export { ModalContentProps } from './ModalContent';
