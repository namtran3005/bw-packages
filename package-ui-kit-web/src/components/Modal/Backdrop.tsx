import React from 'react';
import styled from 'styled-components';
import { Colors, ColorsType } from '@bitwala-cryptobank-squad/package-theme';
import { Backdrop as MuiBackdrop } from '@material-ui/core';

import { BackdropProps } from './types';

interface StyledBackdropProps {
  background?: ColorsType;
  onClick?: () => void;
}

export const StyledMuiBackdrop = styled(MuiBackdrop).withConfig({
  shouldForwardProp: (prop) =>
    prop === 'ref' || !['paperForeground', 'theme'].includes(prop),
})<Omit<StyledBackdropProps, 'theme' | 'onClick'>>`
  &&& {
    background: ${() => Colors.getWithAlpha('backgroundLight', 0.8)};
    inset: 0px;
    pointer-events: initial;
    position: fixed;
    transition: opacity 220ms ease 0s;
    padding: 32px;
  }
`;

export const Backdrop = (props: BackdropProps): React.ReactElement => {
  return (
    <StyledMuiBackdrop
      open={props.open || false}
      background={(props.background as ColorsType) || undefined}
    />
  );
};
