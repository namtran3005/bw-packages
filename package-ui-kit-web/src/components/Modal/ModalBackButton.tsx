import React from 'react';
import { Icons } from '../..';
import { IconButton, IconButtonProps } from '../IconButton';

function ModalBackButton({
  onClose,
}: {
  onClose?: IconButtonProps['onClick'];
}): React.ReactElement {
  return (
    <IconButton onClick={onClose} aria-label="Close">
      <Icons.ArrowLeft />
    </IconButton>
  );
}

export default ModalBackButton;
