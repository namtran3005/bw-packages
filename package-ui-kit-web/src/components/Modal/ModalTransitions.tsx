import * as React from 'react';
import Fade, { FadeProps } from '@material-ui/core/Fade';
import Slide, { SlideProps } from '@material-ui/core/Slide';

const TransitionFade: React.ForwardRefExoticComponent<
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  FadeProps | { children?: any }
> = React.forwardRef((props, ref) => <Fade {...props} ref={ref} />);

const TransitionSlide: React.ForwardRefExoticComponent<SlideProps> = React.forwardRef(
  (props: SlideProps, ref) => <Slide {...props} ref={ref} />
);

export {
  TransitionSlide,
  SlideProps as TransitionSlideProps,
  TransitionFade,
  FadeProps as TransitionFadeProps,
};
