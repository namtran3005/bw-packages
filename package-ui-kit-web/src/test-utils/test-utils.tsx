import {
  render as rtlRender,
  waitFor,
  RenderOptions as RTLRenderOptions,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { WebAppThemeProvider } from '../theme';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

type RenderOptions = RTLRenderOptions;

function renderWithTestHarness(
  ui: React.ReactElement,
  renderOptions: RenderOptions = {}
) {
  function Wrapper({ children }: any) {
    return (
      <MemoryRouter>
        <WebAppThemeProvider>{children}</WebAppThemeProvider>
      </MemoryRouter>
    );
  }

  return rtlRender(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

// Use if you're in a hurry (better than nothing)
// be sure to verify no memory leak from implicit
// waitFor by adding these flags to jest: --runInBand --detectOpenHandles
async function renderAct(ui: React.ReactElement, options: RenderOptions = {}) {
  const view = renderWithTestHarness(ui, options);
  return await waitFor(() => view, {
    timeout: 5000,
    onTimeout: (e) => {
      // eslint-disable-next-line no-console
      console.error(e);
      return e;
    },
  });
}

export * from '@testing-library/react';

export { renderWithTestHarness, renderAct, userEvent };
