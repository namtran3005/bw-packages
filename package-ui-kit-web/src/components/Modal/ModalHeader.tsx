import React, { ReactNode } from 'react';
import styled from 'styled-components';

import { Colors } from '@bitwala-cryptobank-squad/package-theme';

import { IconButtonProps } from '../IconButton';
import { Typography, TypographyProps } from '../Typography';
import { ModalProps } from './types';
import ModalBackButton from './ModalBackButton';

const StyledModalHeader = styled.div`
  &&& {
    z-index: 1;
    display: flex;
    justify-content: space-around;
    flex-direction: column;
    width: 100%;
    height: 76px;
    background-color: ${Colors.get('white')};
    padding: 0;

    > .MuiButtonBase-root {
      width: auto !important;
    }

    > button {
      position: absolute;
      width: auto;
      height: auto;
      padding: 0;
      margin: 0 !important;
    }
    > button:hover {
      background-color: unset;
    }
  }

  .right-header-element {
    position: absolute;
    right: 5%;
  }
`;

const StyledModalTitle = styled(Typography.SmallBody).withConfig({
  shouldForwardProp: (
    prop: keyof (TypographyProps & Pick<ModalProps, 'doNotRestrictTitleWidth'>)
  ) => prop !== 'doNotRestrictTitleWidth',
})`
  &&& {
    color: ${Colors.get('primaryBlack')};
    font-size: 14px;
    height: fit-content;
    position: relative;
    max-width: ${(props) => (props.doNotRestrictTitleWidth ? 'unset' : '50%')};
    margin: auto;
  }
`;

interface ModalHeaderProps {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ref?: React.ForwardedRef<any>;
  onClose?: IconButtonProps['onClick'];
  title?: ReactNode;
  rightHeaderElement?: ModalProps['rightHeaderElement'];
  showDismissButton?: boolean;
  doNotRestrictTitleWidth?: ModalProps['doNotRestrictTitleWidth'];
}

const ModalHeader: React.FC<ModalHeaderProps> = React.forwardRef(
  (props, ref) => (
    <StyledModalHeader ref={ref}>
      {props.showDismissButton && <ModalBackButton onClose={props.onClose} />}
      {props.title && (
        <StyledModalTitle
          textAlign="center"
          doNotRestrictTitleWidth={props.doNotRestrictTitleWidth}
        >
          {props.title}
        </StyledModalTitle>
      )}
      {props.rightHeaderElement && (
        <span className="right-header-element">{props.rightHeaderElement}</span>
      )}
    </StyledModalHeader>
  )
);

export default ModalHeader;
