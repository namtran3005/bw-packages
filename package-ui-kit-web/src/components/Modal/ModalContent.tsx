import React, { ReactNode } from 'react';
import styled from 'styled-components';
import { Spacing } from '@bitwala-cryptobank-squad/package-theme';
import {
  Root as StyledContent,
  RootProps as ContentProps,
  RootContent as StyledContentContent,
} from '../Content/Content.styled';
import ModalHeader from './ModalHeader';
import { ModalProps } from './types';

interface ModalContentProps extends ContentProps {
  verticalAlign?: string;
  docked?: string;
  onClose?: ModalProps['onClose'];
  title?: ReactNode;
  rightHeaderElement?: ModalProps['rightHeaderElement'];
  showDismissButton?: boolean;
  doNotRestrictTitleWidth?: ModalProps['doNotRestrictTitleWidth'];
}

// TODO: when migrating the content component, just add
//       support for ref access and this can be removed
// eslint-disable-next-line @typescript-eslint/no-explicit-any
const Content: React.ForwardRefExoticComponent<any> = React.forwardRef(
  function Content(props: ContentProps, ref) {
    const { children, ...cleanProps } = props;
    return (
      <StyledContent ref={ref} {...cleanProps}>
        <StyledContentContent>{children}</StyledContentContent>
      </StyledContent>
    );
  }
);

const StyledModalContent = styled(Content).withConfig({
  shouldForwardProp: (prop) =>
    !['docked', 'title', 'rightHeaderElement'].includes(prop.toString()),
})`
  && {
    height: 100%;
    width: 100%;
    max-width: 100%;
    padding: 0;
    text-align: center;
    & > div {
      position: relative;
      display: flex;
      flex-direction: column;
      padding-top: 0;
      min-height: calc(100% - ${Spacing.base}px);
      box-sizing: content-box;
    }
  }
  & .MuiGrid-container {
    width: 100%;
    margin: 0;
  }

  a,
  p,
  h1,
  h2,
  h3,
  h4,
  h5 {
    text-align: center;
  }
`;

const ModalContent: React.ForwardRefExoticComponent<ModalContentProps> = React.forwardRef(
  (props, ref) => {
    const {
      children,
      showDismissButton,
      doNotRestrictTitleWidth,
      ...cleanProps
    } = props;
    return (
      <StyledModalContent ref={ref} spacing={24} {...cleanProps}>
        <ModalHeader
          title={props.title}
          onClose={props.onClose}
          rightHeaderElement={props.rightHeaderElement}
          showDismissButton={showDismissButton}
          doNotRestrictTitleWidth={doNotRestrictTitleWidth}
        />
        {children}
      </StyledModalContent>
    );
  }
);

export { ModalContent, ModalContentProps };
