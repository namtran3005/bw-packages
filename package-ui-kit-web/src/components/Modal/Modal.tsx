import React from 'react';
import styled, { css } from 'styled-components';
import { Dialog as MuiDialog } from '@material-ui/core';
import { TransitionProps as MuiTransitionProps } from '@material-ui/core/transitions/transition';
import { Colors } from '@bitwala-cryptobank-squad/package-theme';

import { shadows } from '../Paper';
import { ModalContent } from './ModalContent';
import { TransitionFade, TransitionSlide } from './ModalTransitions';
import { ModalProps, TransitionProps } from './types';
import { StyledMuiBackdrop } from './Backdrop';

export const modalPaperStyle = ({
  docked,
  paperForeground,
  maxPaperWidth,
  doNotStretchContentHeight,
}: Pick<
  ModalProps,
  'docked' | 'paperForeground' | 'maxPaperWidth' | 'doNotStretchContentHeight'
>) => css`
  position: relative;
  width: 100%;
  max-width: ${maxPaperWidth || '423px'};
  height: ${doNotStretchContentHeight ? 'unset' : 'calc(100% - 64px)'};
  ${docked === 'right' && 'right: 32px;'}
  ${docked === 'left' && 'left: 32px;'}
  background: ${Colors.get(paperForeground || 'white')};
  margin-right: ${docked === 'right' ? 0 : 'auto'};
  margin-left: ${docked === 'left' ? 0 : 'auto'};
  align-self: ${docked && !doNotStretchContentHeight ? 'stretch' : 'auto'};
  box-shadow: ${shadows[24]};

  &:focus {
    outline: none;
  }
`;

delete MuiDialog['propTypes'];

export const StyledMuiDialog: React.FC<
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ModalProps | { TransitionProps: any }
> = styled(MuiDialog).withConfig({
  shouldForwardProp: (prop) =>
    prop === 'ref' ||
    !['paperForeground', 'maxPaperWidth', 'doNotStretchContentHeight'].includes(
      prop
    ),
})`
  &&& {
    display: flex;
    padding: 0 !important;

    &:focus {
      outline: none;
    }

    .scrollPaper {
      width: 100%;
      box-sizing: border-box;
      position: relative;
    }
    .paper {
      ${modalPaperStyle}
    }
    .MuiDialog-paperScrollPaper {
      border-radius: 8px;
    }
  }
`;

const Slide = React.forwardRef(function Slide(props: TransitionProps, ref) {
  const { docked, ...cleanProps } = props;
  return (
    <TransitionSlide
      direction={docked === 'right' ? 'left' : 'right'}
      {...cleanProps}
      ref={ref}
    />
  );
});

const Fade = React.forwardRef(function Transition(
  props: MuiTransitionProps,
  ref
) {
  return <TransitionFade {...props} ref={ref} />;
});

const TransitionComponent = React.forwardRef((props: TransitionProps, ref) =>
  props.docked === 'middle' ? (
    <Fade {...props} ref={ref} />
  ) : (
    <Slide {...props} ref={ref} />
  )
);

export function Modal({
  children,
  title,
  onClose,
  ariaLabel,
  docked = 'right',
  paperForeground = 'white',
  rightHeaderElement,
  showDismissButton = true,
  open = false,
  maxPaperWidth,
  doNotRestrictTitleWidth,
  doNotStretchContentHeight,
}: ModalProps): React.ReactElement {
  // allow the fadeout transition to finish
  const [display, setDisplay] = React.useState<boolean>(false);

  React.useEffect(() => {
    let timer: ReturnType<typeof setTimeout> | null | undefined = undefined;

    if (open && !display) {
      timer = undefined;
      setDisplay(true);
    } else if (!open && display) {
      timer = setTimeout(() => setDisplay(false), 300);
    }

    return () => {
      if (timer) {
        clearTimeout(timer);
      }
    };
  }, [display, setDisplay, open]);

  return (
    <React.Fragment>
      {display && (
        <StyledMuiDialog
          aria-label={ariaLabel || title || 'popup-dialog'}
          BackdropProps={{ open }}
          onBackdropClick={onClose}
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
          BackdropComponent={StyledMuiBackdrop as any}
          open={display}
          TransitionComponent={
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            TransitionComponent as React.ForwardRefExoticComponent<any>
          }
          TransitionProps={
            {
              docked,
              in: open,
              enter: open,
              exit: !open,
              timeout: { enter: 300, exit: 300 },
              // eslint-disable-next-line @typescript-eslint/no-explicit-any
            } as any
          }
          docked={docked}
          classes={{
            root: 'root',
            scrollPaper: 'scrollPaper',
            scrollBody: 'scrollBody',
            paper: 'paper',
            paperScrollPaper: 'paperScrollPaper',
            paperScrollBody: 'paperScrollBody',
            paperFullWidth: 'paperFullWidth',
            paperFullScreen: 'paperFullScreen',
          }}
          paperForeground={paperForeground}
          closeAfterTransition
          maxPaperWidth={maxPaperWidth}
          doNotStretchContentHeight={doNotStretchContentHeight}
        >
          <ModalContent
            title={title}
            onClose={onClose}
            docked={docked}
            rightHeaderElement={rightHeaderElement}
            showDismissButton={showDismissButton}
            doNotRestrictTitleWidth={doNotRestrictTitleWidth}
          >
            {children}
          </ModalContent>
        </StyledMuiDialog>
      )}
    </React.Fragment>
  );
}
