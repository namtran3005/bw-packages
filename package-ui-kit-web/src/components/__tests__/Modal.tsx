import * as React from 'react';

import { renderWithTestHarness } from '../../test-utils/test-utils';
import { Modal, ModalProps } from '../Modal';

const defaultModalProps: ModalProps = { open: true };

describe('Modal', () => {
  it('should render Modal', () => {
    const { baseElement } = renderWithTestHarness(
      <Modal {...defaultModalProps} />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
