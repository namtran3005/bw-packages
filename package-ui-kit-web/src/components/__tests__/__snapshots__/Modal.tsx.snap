// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render Modal 1`] = `
.c2 {
  position: relative;
  margin-left: 0px;
  max-width: none;
  min-width: none;
  box-sizing: border-box;
  height: auto;
  min-height: auto;
  z-index: initial;
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
}

.c2 > .{
  box-sizing: border-box;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: inherit;
  margin-right: inherit;
  max-width: auto;
  text-align: inherit;
  height: auto;
  -webkit-transition: 300ms cubic-bezier(.62,.28,.23,.99);
  transition: 300ms cubic-bezier(.62,.28,.23,.99);
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 24px;
  padding-right: 24px;
}

.c5.c5.c5 {
  color: #2C232E;
  box-sizing: content-box;
  padding: 13px;
  width: 22px;
  height: 22px;
  -webkit-transition: background-color 300ms cubic-bezier(.62,.28,.23,.99);
  transition: background-color 300ms cubic-bezier(.62,.28,.23,.99);
  background: transparent;
}

.c5.c5.c5:hover,
.c5.c5.c5:focus {
  background-color: #F0F0F0;
}

.c5.c5.c5:active {
  background-color: rgba(44,35,46,0.25);
}

.c4.c4.c4 {
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 76px;
  background-color: #FFFFFF;
  padding: 0;
}

.c4.c4.c4 > .MuiButtonBase-root {
  width: auto !important;
}

.c4.c4.c4 > button {
  position: absolute;
  width: auto;
  height: auto;
  padding: 0;
  margin: 0 !important;
}

.c4.c4.c4 > button:hover {
  background-color: unset;
}

.c4 .right-header-element {
  position: absolute;
  right: 5%;
}

.c3.c3 {
  height: 100%;
  width: 100%;
  max-width: 100%;
  padding: 0;
  text-align: center;
}

.c3.c3 > div {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 0;
  min-height: calc(100% - 24px);
  box-sizing: content-box;
}

.c3 .MuiGrid-container {
  width: 100%;
  margin: 0;
}

.c3 a,
.c3 p,
.c3 h1,
.c3 h2,
.c3 h3,
.c3 h4,
.c3 h5 {
  text-align: center;
}

.c1.c1.c1 {
  background: rgba(240,240,240,0.8);
  inset: 0px;
  pointer-events: initial;
  position: fixed;
  -webkit-transition: opacity 220ms ease 0s;
  transition: opacity 220ms ease 0s;
  padding: 32px;
}

.c0.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 !important;
}

.c0.c0.c0:focus {
  outline: none;
}

.c0.c0.c0 .scrollPaper {
  width: 100%;
  box-sizing: border-box;
  position: relative;
}

.c0.c0.c0 .paper {
  position: relative;
  width: 100%;
  max-width: 423px;
  height: calc(100% - 64px);
  right: 32px;
  background: #FFFFFF;
  margin-right: 0;
  margin-left: auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  box-shadow: 0px 11px 15px -7px rgba(0,0,0,0.04000000000000001),0px 24px 38px 3px rgba(0,0,0,0.028000000000000004),0px 9px 46px 8px rgba(0,0,0,0.024);
}

.c0.c0.c0 .paper:focus {
  outline: none;
}

.c0.c0.c0 .MuiDialog-paperScrollPaper {
  border-radius: 8px;
}

<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    aria-label="popup-dialog"
    class="MuiDialog-root root c0"
    docked="right"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root c1"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper scrollPaper"
      role="none presentation"
      style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 300ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 300ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiPaper-root MuiDialog-paper paper MuiDialog-paperScrollPaper paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
        role="dialog"
      >
        <div
          class="c2 c3"
        >
          <div
            class=""
          >
            <div
              class="c4"
            >
              <button
                aria-label="Close"
                class="MuiButtonBase-root MuiIconButton-root c5"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    color="currentColor"
                    fill="none"
                    height="32"
                    stroke-width="1.6"
                    viewBox="0 0 32 32"
                    width="32"
                  >
                    <path
                      d="M2 15.333h28M6.667 10.667L2 15.333 6.667 20"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1.6"
                      vector-effect="non-scaling-stroke"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
